src/instrument.cpp

```diff
@@ -300,19 +343,153 @@ void BytecodeRewriter::rewriteCode() {
         u16 end_pc = get16();
         u16 handler_pc = get16();
         u16 catch_type = get16();
-        put16(EXTRA_BYTECODES + start_pc);
-        put16(EXTRA_BYTECODES + end_pc);
-        put16(EXTRA_BYTECODES + handler_pc);
+        put16(start_pc + relocation_table[start_pc]);
+        put16(start_pc + relocation_table[end_pc]);
+        put16(start_pc + relocation_table[handler_pc]);
         put16(catch_type);
     }
 
-    rewriteAttributes(SCOPE_REWRITE_CODE);
+    rewriteCodeAttributes(relocation_table);
+    delete[] relocation_table;
 
     // Patch attribute length
     *(u32*)(_dst + code_begin - 4) = htonl(_dst_len - code_begin);
 }
 
-void BytecodeRewriter::rewriteBytecodeTable(int data_len) {
+// Return the relocation after the last byte of code
+u32 BytecodeRewriter::rewriteCodeForLatency(const u8* code, u32 code_length, u32* relocation_table) {
+    // First scan: identify the maximum possible relocation for any position in code[]
+    // Relocation can happen due to:
+    // - Adding a new invocation
+    // - Narrow jump becoming a wide jump
+    u32 max_relocation = 0;
+    for (u32 i = 0; i < code_length;) {
+        u8 opcode = code[i];
+        if (isFunctionExit(opcode)) {
+            max_relocation += EXTRA_BYTECODES;
+        } else if (isNarrowJump(opcode)) {
+            max_relocation += 2;
+        }
+        i += OPCODE_LENGTH[opcode];
```

Comment:
`wide` bytecode (0xc4) needs special handling (including all other places where you rely on `OPCODE_LENGTH`).

Reply:
Actually, we need special handling for `lookupswitch` and `tableswitch` too: 0159b9cc0543adab65fb0d162b9875429512bcfb